src/Common/Core/Impl/Shell/IIdleTimeService.cs

```diff
@@ -2,8 +2,10 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 
+using System.Threading;
+
 namespace Microsoft.Languages.Editor.Host {
     public interface IIdleTimeService {
-        void DoIdle();
+        void DoIdle(Thread thread = null);
```

Comment:
When tests run in parallel while EditorShell is a singleton, they may get events fired from a wrong thread. In most cases it doesn't matter though, but I've seen some fail occasionally. However, with changes where editor objects are disposed properly, this may not be that necessary.